lib/improver/tests/nowcasting/nowcasting/test_ApplyOrographicEnhancement.py

```diff
@@ -0,0 +1,457 @@
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.51
+        result = plugin._extract_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.second_slice.metadata)
+        self.assertEqual(result, self.second_slice)
+        self.assertEqual(result.coord("time"), self.second_slice.coord("time"))
+
+    def test_alternative_time_quarter_to(self):
+        """Test extracting a time coordinate from the orographic enhancement
+        cube at quarter to an hour."""
+        plugin = ApplyOrographicEnhancement("add")
+        self.precip_cubes[0].coord("time").points = 412227.75
+        result = plugin._extract_orographic_enhancement_cube(
+            self.precip_cubes[0], self.oe_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.second_slice.metadata)
+        self.assertEqual(result, self.second_slice)
+        self.assertEqual(result.coord("time"), self.second_slice.coord("time"))
+
+
+class Test__apply_cube_combiner(IrisTest):
+
+    """Test the __apply_cube_combiner method."""
+
+    def setUp(self):
+        """Set up cubes for testing."""
+        self.precip_cubes = set_up_precipitation_rate_cube()
+        self.oe_cubes = set_up_orographic_enhancement_cube()
+
+    def test_check_expected_values_first_add(self):
+        """Test the expected values are returned when cubes are added.
+        First check."""
+        expected0 = np.array([[[[0., 1., 2.],
+                                [1., 2., 3.],
+                                [0., 3., 4.]]]])
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin._apply_cube_combiner(
+            self.precip_cubes[0], self.oe_cubes[0])
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.precip_cubes[0].metadata)
+        result.convert_units("mm/hr")
+        self.assertArrayAlmostEqual(result.data, expected0)
+
+    def test_check_expected_values_second(self):
+        """Test the expected values are returned when cubes are combined.
+        Second check."""
+        expected1 = np.array([[[[6., 5., 1.],
+                                [6., 5., 1.],
+                                [6., 5., 1.]]]])
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin._apply_cube_combiner(
+            self.precip_cubes[1], self.oe_cubes[1])
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.metadata, self.precip_cubes[1].metadata)
+        result.convert_units("mm/hr")
+        self.assertArrayAlmostEqual(result.data, expected1)
+
+    def test_check_scalar_time_dimensions(self):
+        """Test the expected values are returned when cubes are combined when
+        the dimensions of the input cubes mismatch."""
```

Comment:
I don't quite understand what's going on here.  If the input cube dimensions don't match the plugin needs to throw an error.  If that's a misinterpretation, please could you clarify this docstring.

Reply:
I've extended this docstring to try to explain why this situation often arises, and isn't an error, as the extraction of a particular time point with `_select_orographic_enhancement cube` can yield a cube with a scalar coordinate, whereas the input precipitation cube could have time as a dimension coordinate. This is corrected in `_apply_orographic_enhancement` using check_cube_coordinates, so this test is to ensure that any mismatching in coordinates e.g. whether the time coordinate is a dimension coordinate or a scalar coordinate is corrected.